### PR TITLE
Simplify the Database Configuration

### PR DESCRIPTION
**Pull request by PSchmiedmayer**
# Simplify the Database Configuration

## :recycle: Current situation & Problem
ApodiniDatabase currently provides a wrapper around some possible database configurations that limits the about of possibilities and introduces unnecessary dependences.

## :bulb: Proposed solution
Use the build in FluentKit mechanisms for the database configuration.

## :gear: Release Notes 
Introduces a breaking change for the `DatabaseConfiguration` adding a database databaseID that is passed to FluentKit. E.g.:
```swift
 DatabaseConfiguration(.sqlite(.memory))
```
is changed to
```swift
 DatabaseConfiguration(.sqlite(.memory), as: .sqlite)
```
[Check out the Fluent documentation for more information](https://docs.vapor.codes/4.0/fluent/overview/).

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -163,12 +159,7 @@ let package = Package(
             dependencies: [
                 .target(name: "Apodini"),
                 .target(name: "ApodiniVaporSupport"),
-                .product(name: "Fluent", package: "fluent"),
-                .product(name: "FluentKit", package: "fluent-kit"),
-                .product(name: "FluentMongoDriver", package: "fluent-mongo-driver"),
-                .product(name: "FluentSQLiteDriver", package: "fluent-sqlite-driver"),
```
Comment (bauer-andreas) [inline Package.swift:177]: Removing the SQLLite drivers should allow us to remove the sqlite dependency installs currently happening in the CI right?

**Review by bauer-andreas — APPROVED** [review]
Great change. The PR introduces some future proofing for the `ApdoiniDatabase` package.
Also from a dependency point of view it is only a win in general to no longer depend on the individual drivers as direct dependencies.

I got only a small comment in regards to the CI setup, which could be simplified by removing the install steps for the sqllitedev package.

File: Package.swift (hunk 1 above)
Comment (PSchmiedmayer) [reply Package.swift:177]: Sadly no. We still use the in memory database SQLite driver for the unit tests and it therefore it is still required to have the dependencies installed in the CI environment.

Comment (bauer-andreas) [reply Package.swift:177]: Maybe related to some missing understanding of fluent internals, but we can use the sqlite driver without specifying it as a dependency in our `Package.swift`? I mean, it compiles, so the answer is probably yes 🙃

Comment (PSchmiedmayer) [reply Package.swift:177]: It is still defined as a package dependency (line 46 in Package.swift) but just not a dependency in the ApodiniDatabase target and only in the ApodiniTests target. The dependency is removed here but we still use it in the other test target.

Comment (bauer-andreas) [reply Package.swift:177]: Ah, I completely confused the dependencies section 😅 Thanks for the clarification.

**Pull request merged**